syntax = "proto3";

package transport_guide_catalogue;

import "svg.proto";
import "transport_router.proto";
import "map_renderer.proto";

message Coordinates
{
    double lat = 1;
    double lng = 2;
}

message Stop
{
    int32 id = 1;
    string name_stop = 2;
    Coordinates coordinates = 3;
}

message Bus
{
    string name_bus = 1;
    repeated string stops = 2;
    bool is_roundtrip = 3;
}

message Distance
{
    string name_stop_from = 1;
    string name_stop_to = 2;
    int32 distance = 3;
}

message TransportCatalogue
{
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated Distance distances = 3;
    MapParameter map_parameter = 4;
    RoutingSettings routing_settings = 5;
}